Cannot compile all-clusters-app on M1 Mac

With SHA 6c4b54ec544cf748180c1e32a41f6e4b1b34b7fe

all-clusters-app cannot be complied on on-network device (mac)

**Steps to reproduce the issue;**

1. cd ~/connectedhomeip
2. git reset --hard 6c4b54ec544cf748180c1e32a41f6e4b1b34b7fe
3. source scripts/activate.sh
4. scripts/examples/gn_build_example.sh examples/all-clusters-app/linux out/debug chip_config_network_layer_ble=false

**Error:**

FAILED: address-resolve-tool address-resolve-tool.map 
g++ -target x86_64-apple-macos10.15 -O0 -fPIC -Werror -fdiagnostics-color -Wl,-dead_strip -L/opt/homebrew/Cellar/openssl@3/3.0.3/lib -Wl,-map,./address-resolve-tool.map @./address-resolve-tool.rsp -o ./address-resolve-tool
ld: warning: ignoring file /opt/homebrew/Cellar/openssl@3/3.0.3/lib/libcrypto.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file /opt/homebrew/Cellar/openssl@3/3.0.3/lib/libssl.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
Undefined symbols for architecture x86_64:
  "_RAND_priv_bytes", referenced from:
      chip::Crypto::DRBG_get_bytes(unsigned char*, unsigned long) in libChipCrypto.a(libChipCrypto.CHIPCryptoPALOpenSSL.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[440/455] ar libCHIPAppServer.a
ninja: build stopped: subcommand failed.


Attaching the complete error log:

[on-netwrok_logs.txt](https://github.com/project-chip/connectedhomeip/files/8828048/on-netwrok_logs.txt)

--- a/scripts/bootstrap.sh
+++ b/scripts/bootstrap.sh
@@ -58,11 +58,6 @@ EOF
     PW_ROOT="$_CHIP_ROOT/third_party/pigweed/repo"
     export PW_ROOT
 
-    # Do not force use Rosetta in Pigweed because Matter is using host toolchain
-    if [ -z "$PW_BOOTSTRAP_USE_ROSETTA" ]; then
-        export PW_BOOTSTRAP_USE_ROSETTA=false
-    fi
-
     . "$_CHIP_ROOT/third_party/pigweed/repo/pw_env_setup/util.sh"
 
     _chip_bootstrap_banner() {

--- a/scripts/environment.json
+++ b/scripts/environment.json
@@ -9,5 +9,6 @@
         "gn_targets": [":python_packages.install"]
     },
     "required_submodules": ["third_party/pigweed/repo"],
+    "rosetta": "never",
     "gni_file": "build_overrides/pigweed_environment.gni"
 }

